File: 2. Array and String/9. contains_duplicate/contains_duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
 var containsDuplicate = function(nums) {
    let duplicateMap = {}
    for(let i=0; i<nums.length; i++){
        if(!duplicateMap[nums[i]])
            duplicateMap[nums[i]] = 1
        else
            return true
    }

    return false
};


console.log(containsDuplicate([1,2,3,1]));
console.log(containsDuplicate([1,2,3,4])); 
console.log(containsDuplicate([1,1,1,3,3,4,3,2,4,2]));

File: 2. Array and String/9. contains_duplicate/contains_duplicate.py
class Solution:
    # def containsDuplicate(self, nums) -> bool:
    #     charCountMap = {}
    #     for num in nums:
    #         if(num in charCountMap):
    #             return True
    #         else:
    #             charCountMap[num] = 1

    #     return False

    def containsDuplicate(self, nums) -> bool:
        mySet = set()
        for num in nums:
            if(num in mySet):
                return True
            mySet.add(num) 

        return False

    def containsDuplicate2(self, nums) -> bool:
        mySet = set(nums)
        if(len(nums) != len(mySet)):
            return True
        return False

test = Solution()
print(test.containsDuplicate([1,2,3,1]))
print(test.containsDuplicate([1,2,3,4]))
print(test.containsDuplicate([1,1,1,3,3,4,3,2,4,2]))
print(test.containsDuplicate2([1,2,3,1]))
print(test.containsDuplicate2([1,2,3,4]))
print(test.containsDuplicate2([1,1,1,3,3,4,3,2,4,2]))